### ISSUE TITLE
False Positive for 'customElement: true'.

### ISSUE THREAD

**Issue by irvingoujAtDevolution** [sveltejs/eslint-plugin-svelte#602]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.52.0

### What version of `eslint-plugin-svelte` are you using?

2.34.0

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
	root: true,
	parser: '@typescript-eslint/parser',
	extends: ['eslint:recommended', 'plugin:@typescript-eslint/recommended', 'prettier', 'plugin:svelte/recommended'],
	plugins: ['@typescript-eslint'],
	ignorePatterns: ['*.cjs'],
	overrides: [
		{
			files: ['*.svelte'],
			parser: 'svelte-eslint-parser',
			parserOptions: {
				parser: '@typescript-eslint/parser'
			}
		}
	],
	parserOptions: {
		sourceType: 'module',
		ecmaVersion: 2020,
		project: './tsconfig.json',
		extraFileExtensions: ['.svelte']
	},
	env: {
		browser: true,
		es2017: true,
		node: true
	},
	rules: {
		'@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
		'@typescript-eslint/no-explicit-any': 'off',
		'no-prototype-builtins': 'off',
		// 'svelte/valid-compile':'off'
	},
	settings: {
		svelte: {
			compileOptions: {
				customElement: true
			}
		}
	}
};

```
</details>

```svelte
<svelte:options tag="iron-remote-gui"/>

<script lang="ts">
    import {onMount} from 'svelte';
...
</script>
...

```

### What did you expect to happen?

I expect eslint throw no error with run `npx eslint src/**`

### What actually happened?

```
> npx eslint src/**


C:\Users\jou\code\IronRDP\web-client\iron-remote-gui\src\iron-remote-gui.svelte
  1:17  error  The 'tag' option is used when generating a custom element. Did you forget the 'customElement: true' compile option?(missing-custom-element-compile-options)  svelte/valid-compile

✖ 1 problem (1 error, 0 warnings)

```

### Link to **GitHub Repo** with Minimal Reproducible Example

[IronRDP](https://github.com/Devolutions/IronRDP/pull/258/files#diff-c24a4bae4bdc3978d2fdbe7729ffeb77a49509e8405d8dc2d966db2df8306e58)

inside the ./web-client/iron-remote-gui, one can try remove the
```
svelte/valid-compile': 'off'
```
to reproduce this problem

**Comment by baseballyama** [comment]
This error is came from Svelte compiler, so I think this is not ESLint issue.
Can you try to compile your code with Svelte compilar and share the result?

**Comment by irvingoujAtDevolution** [comment]
I use vite build and it works well. Everything compiles.

**Comment by irvingoujAtDevolution** [comment]
here is my vite config
```
import { defineConfig } from 'vite'
import { svelte } from '@sveltejs/vite-plugin-svelte'
import wasm from "vite-plugin-wasm";
import topLevelAwait from "vite-plugin-top-level-await";
import dtsPlugin from "vite-plugin-dts";

// https://vitejs.dev/config/
export default defineConfig({
  build: {
    lib: {
      entry: './src/main.ts',
      name: 'IronRemoteGui',
      formats: ["umd", "es"],
    }
  },
  server: {
    fs: {
      strict: false
    }
  },
  plugins: [svelte({
    compilerOptions:{
      customElement:true
    }
  }), wasm(), topLevelAwait(), dtsPlugin()],
})

```

**Comment by baseballyama** [comment]
It should be fix in 2.34.1!
